source/adapters/hip/kernel.hpp

```diff
@@ -42,6 +42,10 @@ struct ur_kernel_handle_t_ {
   ur_program_handle_t Program;
   std::atomic_uint32_t RefCount;
 
+  static constexpr uint32_t ReqdThreadsPerBlockDimensions = 3u;
+  std::unordered_map<ur_device_handle_t, std::array<size_t, 3>>
```

Comment:
Why do we need this? A `ur_kernel_handle_t_` is unique to a program which is unique to a device.

Reply:
The way it's set up on [Cuda](https://github.com/oneapi-src/unified-runtime/blob/b98584f742226278a79548a0583e715bf890ef1e/source/adapters/cuda/kernel.hpp#L144C28-L144C48) is through a call to `urKernelGetGroupInfo` which takes a `device`. Now that we have multi-device context I though we would need to keep a per-device mapping. If it's safe to always use the first device, we could mimic what Cuda does and have it as a single array of 3 values.

Reply:
It does indeed take a device, but that doesn't mean we need to use it. Every kernel is associated with a single device so we don't need this map. The map adds unnecessary complexity as well as overhead.

Reply:
So should it always use the first available device in the context?

Reply:
We've discussed this offline, done in: 3b8d6c5328cf982f4adf17e3f4140e6d1960b9b9